.github/workflows/build.yml

```diff
@@ -11,26 +11,60 @@ jobs:
   build:
     name: Build binaries
     runs-on: ubuntu-latest
+    outputs:
+      version_major: ${{ steps.build_info.outputs.version_major }}
+      version_minor: ${{ steps.build_info.outputs.version_minor }}
+      version_build: ${{ steps.build_info.outputs.version_build }}
+      image_tag: ${{ steps.build_info.outputs.image_tag }}
+
     steps:
     - name: Checkout code
       uses: actions/checkout@v4
 
+    - id: build_info
+      name: Declare build info
+      run: |
+        version_major=''
+        version_minor=''
+        version_build=''
+        image_tag=''
+
+        branch=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}
+        ref=${{ github.ref }}
+        if [[ "$ref" =~ 'refs/tags/' ]]; then
+          version=$(sed -E 's/^v([0-9.]*)$/\1' <<<$ref )
+          version_major=$(cut -d. -f1 <<<$version)
+          version_minor=$(cut -d. -f2 <<<$version)
+          version_build=$(cut -d. -f3 <<<$version)
+          image_tag="$version"
+        elif [[ "$ref" =~ 'refs/heads/' ]]; then
+          image_tag="${branch}-head"
+        elif [[ "$ref" =~ 'refs/pull/' ]]; then
+          image_tag="pr${{ github.event.number }}"
+        fi
+
+        echo "version_major=${version_major}" >>$GITHUB_OUTPUT
+        echo "version_minor=${version_minor}" >>$GITHUB_OUTPUT
+        echo "version_build=${version_build}" >>$GITHUB_OUTPUT
+        echo "image_tag=${image_tag}" >>$GITHUB_OUTPUT
+
+        cat <<EOF
+        version_major=${version_major}
+        version_minor=${version_minor}
+        version_build=${version_build}
+        image_tag=${image_tag}
+        EOF
+
     # Build binaries
     - name: Run ci
-      run: make ci
+      run: make ci-workflow
```

Comment:
This new makefile target excludes the image building to avoid out-of-space error in the Github builder VM.